From dc4533207c15661d2b336250de6749290cca8f00 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:19:33 -0400
Subject: [fs] nfs: replace count*size kzalloc by kcalloc

Message-id: <1411050057-12417-11-git-send-email-steved@redhat.com>
Patchwork-id: 96188
O-Subject: [RHEL7.1 PATCH 10/94] FS/NFS: replace count*size kzalloc by kcalloc
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Fabian Frederick <fabf@skynet.be>

kcalloc manages count*sizeof overflow.

Cc: Trond Myklebust <trond.myklebust@primarydata.com>
Cc: linux-nfs@vger.kernel.org
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id:f15b5041440c502726f121b37cf3923562f8dc6f
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/blocklayout/blocklayout.c b/fs/nfs/blocklayout/blocklayout.c
index bf42e6a..f35e072 100644
--- a/fs/nfs/blocklayout/blocklayout.c
+++ b/fs/nfs/blocklayout/blocklayout.c
@@ -1069,7 +1069,7 @@ nfs4_blk_get_deviceinfo(struct nfs_server *server, const struct nfs_fh *fh,
   return ERR_PTR(-ENOMEM);
  }
 
- pages = kzalloc(max_pages * sizeof(struct page *), GFP_NOFS);
+ pages = kcalloc(max_pages, sizeof(struct page *), GFP_NOFS);
  if (pages == NULL) {
   kfree(dev);
   return ERR_PTR(-ENOMEM);
diff --git a/fs/nfs/filelayout/filelayoutdev.c b/fs/nfs/filelayout/filelayoutdev.c
index b52cb37..5d94b22 100644
--- a/fs/nfs/filelayout/filelayoutdev.c
+++ b/fs/nfs/filelayout/filelayoutdev.c
@@ -695,7 +695,7 @@ filelayout_get_device_info(struct inode *inode,
  if (pdev == NULL)
   return NULL;
 
- pages = kzalloc(max_pages * sizeof(struct page *), gfp_flags);
+ pages = kcalloc(max_pages, sizeof(struct page *), gfp_flags);
  if (pages == NULL) {
   kfree(pdev);
   return NULL;
-- 
1.7.1